docs/DeveloperGuide.md

```diff
@@ -6,33 +6,360 @@
+ Sorry, an error was encountered! Here is the error description:
+ The start time cannot be the same as the end time!
+______________________________________________________________________
+```
+
+#### Listing appointments
+
+1. Prerequisites: Add an appointment using any of the add commands. At least one appointment in the list.
+2. Test case: `list`<br>
+   Expected output: A list of all previously added appointments.
+   Example:
+```
+______________________________________________________________________
+1.  UID: C84120821A
+ Date: 2023-12-12 | Time: 19:00
+ Animal Name: Lulu | Animal Type: Cat
+ Owner Name: Jon | Contact Number: 91919191
+2.  UID: V12400172X | vaccine: Covid
+ Date: 2023-12-12 | Time: 19:00
+ Animal Name: Russ | Animal Type: Dog
+ Owner Name: Abel | Contact Number: 92929292
+3.  UID: S14837641R | Priority: HIGH
+ Animal Name: Polly | Animal Type: Hamster
+ Owner Name: Sam | Contact Number: 93939393
+ Start Date: 2023-12-12 | Start Time: 19:00
+ End Date: 2023-12-12 | End Time: 20:00
+______________________________________________________________________
+```
+
+#### Finding appointments
+
+1. Prerequisites: Add an appointment using any of the add commands. At least one appointment in the list.
+2. Test case: `find a/Dog`
+   Expected output: A list of all previously added appointments with the animal type.
+   Example:
+```
+______________________________________________________________________
+1.  UID: V12400172X | vaccine: Covid
+ Date: 2023-12-12 | Time: 19:00
+ Animal Name: Russ | Animal Type: Dog
+ Owner Name: Abel | Contact Number: 92929292
+______________________________________________________________________
+```
+
+3. Test case: `find t/consultation`
+   Expected output: A list of all previously added appointments with the appointment type.
+   Example:
+```
+______________________________________________________________________
+1.  UID: C84422868K
+ Date: 2023-12-12 | Time: 19:00
+ Animal Name: Lulu | Animal Type: 
+ Owner Name: Jon | Contact Number: 91919191
+______________________________________________________________________
+```
+
+#### Removing appointments
+
+1. Prerequisites: The UID of the appointment you want to remove. Use `list` or `find` to help you get the specific UID.
+2. Test case: `remove C84120821A`<br>
+   Expected output: The appointment with the specified UID is removed, details shown in the status message.
+   Example:
+```
+______________________________________________________________________
+ This appointment has been removed your appointment manager: 
+ UID: C84120821A
+ Date: 2023-12-12 | Time: 19:00
+ Animal Name: Lulu | Animal Type: Cat
+ Owner Name: Jon | Contact Number: 91919191
+ Task removed successfully!
+______________________________________________________________________
+```
+
+#### Mark/Unmark appointments
+
+1. Prerequisites: The UID of the appointment you want to remove. Use `list` or `find` to help you get the specific UID.
```

Comment:
Earlier it was stated that you did not want the user to manually type in the UID string themselves. I understand that you can use list or find to assist one in finding the UID to help delete the appointment, but would it be better to allow for someone to delete an appointment by name? 
If this is implemented you could prevent duplicate appoint names from being created.